### PR TITLE
SNOW-2043938 Bugfix - RECORD_METADATA.offset stored as integer

### PR DESCRIPTION
**Pull request by sfc-gh-mbobowski**
# Overview

SNOW-2043938

Kafka offset is long not integer, so the connector may hit the overflow.

### CODE REVIEW

**Review by sfc-gh-dseweryn — APPROVED** [review]

**Review by sfc-gh-wtrefon — CHANGES_REQUESTED** [review]
please add test for the overflow of int offset

File: src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 public class IcebergIngestionSchemaEvolutionIT extends IcebergIngestionIT {
 
   private static final String RECORD_METADATA_TYPE =
-      "OBJECT(offset NUMBER(10,0), topic VARCHAR(16777216), partition NUMBER(10,0), key"
+      "OBJECT(offset NUMBER(19,0), topic VARCHAR(16777216), partition NUMBER(10,0), key"
```
[the remark this answers is not in the document]
Comment (sfc-gh-mbobowski) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionSchemaEvolutionIT.java:29]: Sure, test added.

**Review by sfc-gh-wtrefon — APPROVED** [review]
